pkg/controller/common/helpers.go

```diff
@@ -554,47 +556,30 @@ func removeIgnDuplicateFilesUnitsUsers(ignConfig ign2types.Config) (ign2types.Co
 	return ignConfig, nil
 }
 
-// TranspileCoreOSConfigToIgn transpiles Fedora CoreOS config to ignition
-// internally it transpiles to Ign spec v3 config
-func TranspileCoreOSConfigToIgn(files, units []string) (*ign3types.Config, error) {
+// ConvertButaneFragmentsToIgnition takes as input butane fragments (fcos 1.3) and returns an Ignition v3.2 configuration.
+func ConvertButaneFragmentsToIgnition(files, units []string) (*ign3types.Config, error) {
 	overwrite := true
-	outConfig := ign3types.Config{}
+	buCfg := fcosbutane.Config{}
 	// Convert data to Ignition resources
 	for _, d := range files {
-		f := new(fcctbase.File)
+		f := new(bubase.File)
 		if err := yaml.Unmarshal([]byte(d), f); err != nil {
-			return nil, fmt.Errorf("failed to unmarshal file into struct: %v", err)
+			return nil, fmt.Errorf("failed to unmarshal %q into struct: %v", d, err)
 		}
 		f.Overwrite = &overwrite
 
-		// Add the file to the config
-		var ctCfg fcctbase.Config
-		ctCfg.Storage.Files = append(ctCfg.Storage.Files, *f)
-		ign3_0config, tSet, err := ctCfg.ToIgn3_0()
-		if err != nil {
-			return nil, fmt.Errorf("failed to transpile config to Ignition config %s\nTranslation set: %v", err, tSet)
-		}
-		ign3_2config := translate3.Translate(translate3_1.Translate(ign3_0config))
-		outConfig = ign3.Merge(outConfig, ign3_2config)
+		buCfg.Storage.Files = append(buCfg.Storage.Files, *f)
 	}
 
 	for _, d := range units {
-		u := new(fcctbase.Unit)
+		u := new(bubase.Unit)
 		if err := yaml.Unmarshal([]byte(d), u); err != nil {
 			return nil, fmt.Errorf("failed to unmarshal systemd unit into struct: %v", err)
 		}
 
-		// Add the unit to the config
-		var ctCfg fcctbase.Config
-		ctCfg.Systemd.Units = append(ctCfg.Systemd.Units, *u)
-		ign3_0config, tSet, err := ctCfg.ToIgn3_0()
-		if err != nil {
-			return nil, fmt.Errorf("failed to transpile config to Ignition config %s\nTranslation set: %v", err, tSet)
-		}
-		ign3_2config := translate3.Translate(translate3_1.Translate(ign3_0config))
-		outConfig = ign3.Merge(outConfig, ign3_2config)
+		buCfg.Systemd.Units = append(buCfg.Systemd.Units, *u)
 	}
-
+	outConfig, _, _ := buCfg.ToIgn3_2(bucommon.TranslateOptions{})
```

Comment:
is there no way to get an error from this?

Reply:
Nice catch, thank you!